cmd/shfmt/main.go

```diff
@@ -30,8 +31,9 @@ var (
 	find    = flag.Bool("f", false, "")
 	diffOut = flag.Bool("d", false, "")
 
-	langStr = flag.String("ln", "", "")
-	posix   = flag.Bool("p", false, "")
+	langStr    = flag.String("ln", "", "")
+	posix      = flag.Bool("p", false, "")
+	ignoreFile = flag.String("if", ".shfmtignore", "")
```

Comment:
The default doesn't make a lot of sense to me; the point of gitignore files is generally that they work in nested directories too.

Also, why is a flag necessary? If the file exists, the tool should use it.

Reply:
The flag allow use to specify an another ignore file model such as `.gitignore`:

```
shfmt --ignore-path .gitignore
```

Ref: https://github.com/mvdan/sh/issues/288#issuecomment-535381642

It also solve what you said in: https://github.com/mvdan/sh/issues/288#issuecomment-535918319

> it adds a custom -if flag and a custom .shfmtignore file on disk.

It add a custom flag, but not necessary a new ignore file, depending of your choice.

This is the more flexible way and it is working great. But I can also make it less restrictive by just using `.gitignore` directly and without flag. But it will remove flexibility and will not work with other vcs.

I recommend you to take a look to the other lint tools, many of them use this method. This is quite handful, especially for CI.

Reply:
> the point of gitignore files is generally that they work in nested directories too.

You may also have `.gitignore` files on nested directory. Also, I set up the code like this because shfmt can have multiple path arguments:

```
shfmt /path/to/project1 /path/to/project2
```

So the flag is like a "model" file to use.

Reply:
I also like this approach as it seems to be the approach that are taking other formatters like stylelint, prettier, ...

For the commit maybe the flag could be `-ip` instead of `-if` to try to mimic `--ignore-path` that seems to be the more common agreement.

Reply:
I was not sure about the name. I can change it but I would have @mvdan final review for this first.